### ISSUE TITLE
-Dgraal.OptWriteMotion=true breaks villagers AI

### ISSUE THREAD

**Issue by AntonIXO** [brucethemoose/Minecraft-Performance-Flags-Benchmarks#17]
Hi,

I am launched my public bettermc(modpack) server and face thats -Dgraal.OptWriteMotion=true on GraalVM 22.3 Java 19 breaks villagers, thats they do not find their work. They became useless. 
Seems like vanilla ai is ok with this option. I do not know which mod breaks it. I've used v11 modpack version.

**Comment by brucethemoose** [comment]
Yeah, optwritemotion used to straight up crash some modpacks in my testing. 


Are you testing `WriteableCodeCache` as well?

**Comment by AntonIXO** [comment]
> Yeah, optwritemotion used to straight up crash some modpacks in my testing. 
> 
> 
> Are you testing `WriteableCodeCache` as well?
> 

Yes i use

**Comment by brucethemoose** [comment]
> Yeah, optwritemotion used to straight up crash some modpacks in my testing.


Ah now that's interesting. WriteableCodeCache improved chunkgen times in my 22.1.0 benches (which is unusual, and most flags dont improve raw throughput like that), but it caused rare crashes in AllTheMods and Valhesia Enhanced Vanilla for some unknown reason.

**Comment by AntonIXO** [comment]
> > Yes I use
> 
> Ah now that's interesting. WriteableCodeCache improved chunkgen times in my 22.1.0 benches (which is unusual, and most flags dont improve chunkgen throughput like that), but it caused rare crashes in AllTheMods and Valhesia Enhanced Vanilla for some unknown reason.

What is crash error in your modpacks?

**Comment by brucethemoose** [comment]
> > > Yes I use
> > 
> > 
> > Ah now that's interesting. WriteableCodeCache improved chunkgen times in my 22.1.0 benches (which is unusual, and most flags dont improve chunkgen throughput like that), but it caused rare crashes in AllTheMods and Valhesia Enhanced Vanilla for some unknown reason.
> 
> What is crash error in your modpacks?

I dunno. I was using the Benchmark.py script in this repo, and if I was doing, say, 7 runs for a benchmark with `WriteableCodeCache` enabled, one of them would just hang while I was away doing something else. It happened infrequently, but consistently with that flag, but the 6 other runs would complete just fine.


Right now the script doesn't record errors, it just tells you if a run crashes, times out or fails to start in the json file.

This was with GraalVM 22.1.0 though, so that testing isnt necessarily relevant anymore.

**Comment by AntonIXO** [comment]
I have new issue now. I am not sure that is graalvm optimizations. Items drops from broken blocks only after player moving.

**Comment by brucethemoose** [comment]
> I have new issue now. I am not sure that is graalvm optimizations. Items drops from broken blocks only after player moving.

You can check by copying the instance and running it with openjdk. Or just switching jdks and opening the same instance, if you can reproduce it consistently.

**Comment by AntonIXO** [comment]
> > I have new issue now. I am not sure that is graalvm optimizations. Items drops from broken blocks only after player moving.
> 
> You can check by copying the instance and running it with openjdk. Or just switching jdks and opening the same instance, if you can reproduce it consistently.

Find out problem source, it was one of new vmp optimizations
